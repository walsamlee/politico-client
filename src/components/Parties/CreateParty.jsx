/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';

class CreateParty extends Component {
  constructor(props) {
    super(props);

    this.state = {
      image: 'https://res.cloudinary.com/walsam/image/upload/v1559210390/png-icon.png',
      partyname: '',
      parthqaddress: '',
    };
  }

  changeHandler = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  submitHandler = (event) => {
    event.preventDefault();
    const token = localStorage.getItem('token');

    if (token) {
      const partyData = {
        logoUrl: this.state.image,
        name: this.state.partyname,
        hqAddress: this.state.parthqaddress,
      };

      this.props.createParty(partyData, token);
    }
  }

  imageUploadHandler = (event) => {
    const cloudinaryUrl = 'https://api.cloudinary.com/v1_1/walsam/image/upload';
    const cloudinaryUploadPreset = 'w4rdqtby';

    const file = event.target.files[0];

    const imageFormData = new FormData();

    imageFormData.append('file', file);
    imageFormData.append('upload_preset', cloudinaryUploadPreset);

    fetch(cloudinaryUrl, {
      method: 'POST',
      body: imageFormData,
    })
      .then(response => response.json())
      .then((response) => {
        console.log('Uploaded image: ', response.url);
        this.setState({ image: response.url });
      })
      .catch(error => error);
  }

  resetHandler = () => {
    this.setState({
      partyname: '',
      parthqaddress: '',
    });
  };

  render() {
    return (
      <div className="dashboard">
        <div className="signup-form">
          <form id="create-party" name="sign-up" onSubmit={this.submitHandler}>
            <table className="form-table">
              <tr>
                <td className="label" />
                <td>
                  <label id="party-logo-label" htmlFor="party">
                    <input id="party-logo" className="party-logo" type="file" accept="image/*" name="partylogo" onChange={this.imageUploadHandler} />
                  </label>
                </td>
              </tr>
              <tr>
                <td className="label" />
                <td>
                  <input id="party-name" type="text" name="partyname" placeholder="Party Name" required onChange={this.changeHandler} value={this.state.partyname} />
                </td>
              </tr>
              <tr>
                <td className="label" />
                <td>
                  <input id="part-hq-address" type="text" name="parthqaddress" placeholder="HQ Address" required onChange={this.changeHandler} value={this.state.parthqaddress} />
                </td>
              </tr>
              <tr>
                <td />
                <td>
                  <input type="submit" className="btn btn-cart" value="Save" />
                  <input type="reset" className="btn btn-cart" value="Reset" onClick={this.resetHandler} />
                </td>
              </tr>
            </table>
          </form>
        </div>
      </div>
    );
  }
}

export default CreateParty;
