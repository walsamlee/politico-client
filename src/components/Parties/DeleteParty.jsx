/* eslint-disable no-alert */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';

class DeleteParty extends Component {
    deleteHandler = (event) => {
      event.preventDefault();
      if (window.confirm('Are you sure you want to delete party?')) {
        this.props.deleteParty(this.props.partyId, localStorage.getItem('token'));
      }
    }

    render() {
      return (
        <button type="submit" className="btn btn-cart" onClick={this.deleteHandler}>Delete</button>
      );
    }
}

export default DeleteParty;
