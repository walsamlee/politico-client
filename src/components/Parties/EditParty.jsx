/* eslint-disable no-alert */
/* eslint-disable prefer-destructuring */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';
import store from '../../store/store';

class EditParty extends Component {
  constructor(props) {
    super(props);

    this.state = {
      id: '',
      image: '',
      partyname: '',
      parthqaddress: '',
    };
  }

  componentDidMount() {
    const searchParts = window.location.search.split('=');
    const partyId = parseInt(searchParts[1], 10);

    const storeParties = store.getState();
    const parties = storeParties.parties.parties;

    const setPartyState = parties.filter(
      partyx => partyx.id === partyId,
    );

    this.setState({
      id: setPartyState[0].id,
      image: setPartyState[0].logoUrl,
      partyname: setPartyState[0].name,
      parthqaddress: setPartyState[0].hqAddress,
    });
  }

  changeHandler = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  submitHandler = (event) => {
    event.preventDefault();

    const partyName = {
      name: this.state.partyname,
    };

    if (window.confirm('Are you sure you want to save changes?')) {
      this.props.editParty(partyName, this.state.id, localStorage.getItem('token'));
    }
  }

  cancelHandler = () => {
    window.location = '/parties';
  };

  render() {
    return (
      <div className="dashboard">
        <div className="signup-form">
          <form name="sign-up" onSubmit={this.submitHandler}>
            <table className="form-table">
              <tbody>
                <tr>
                  <td className="label" />
                  <td>
                    <div className="preview">
                      <img src={this.state.image} alt="preview" />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td className="label" />
                  <td>
                    <input id="party-name" type="text" name="partyname" placeholder="Party Name" required onChange={this.changeHandler} value={this.state.partyname} />
                  </td>
                </tr>
                <tr>
                  <td className="label" />
                  <td>
                    <p className="edit-notification">* party address not editable</p>
                    <input id="part-hq-address" type="text" name="parthqaddress" placeholder="HQ Address" readOnly onChange={this.changeHandler} value={this.state.parthqaddress} />
                  </td>
                </tr>
                <tr>
                  <td />
                  <td>
                    <input type="submit" className="btn btn-cart" value="Save" />
                    <input type="button" className="btn btn-cart" value="Cancel" onClick={this.cancelHandler} />
                  </td>
                </tr>
              </tbody>
            </table>
          </form>
        </div>
      </div>
    );
  }
}

export default EditParty;
