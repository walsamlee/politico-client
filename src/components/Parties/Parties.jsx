/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import Loader from '../../utils/Loader';
import Party from '../Party/Party';

class Parties extends Component {
  componentDidMount() {
    const { fetchParties } = this.props;
    fetchParties();
  }

  render() {
    const { data } = this.props;
    const { parties } = data;
    return (
      <div className="dashboard">
        <div className="edit-board">
          <div className="party-view">
            {
                parties.fetching
                  ? <Loader />
                  : parties.parties.map(party => (
                    <Party key={party.id} party={party} />
                  ))
            }
          </div>
        </div>
      </div>
    );
  }
}

export default Parties;
