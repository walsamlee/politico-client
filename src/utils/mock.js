const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
