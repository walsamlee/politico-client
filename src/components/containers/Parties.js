/* eslint-disable arrow-body-style */
import { connect } from 'react-redux';
import Parties from '../Parties/Parties';
import partyAction from '../../actions/partyAction';

const mapStateToProps = state => ({
  data: state,
});

const matchDispatchToProps = (dispatch) => {
  return {
    fetchParties: () => {
      dispatch(partyAction.fetchParties());
    },
  };
};

const PartiesContainer = connect(
  mapStateToProps,
  matchDispatchToProps,
)(Parties);

export default PartiesContainer;
