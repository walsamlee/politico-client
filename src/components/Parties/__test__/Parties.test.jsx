import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import Parties from '../Parties';

const { partiesData } = mock;


describe('<Parties />', () => {
  it('should render Parties component', () => {
    const wrapper = shallow(
      <Parties data={partiesData} fetchParties={jest.fn()} />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
