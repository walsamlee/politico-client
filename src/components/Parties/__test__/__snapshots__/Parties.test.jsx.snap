// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Parties /> should render Parties component 1`] = `ShallowWrapper {}`;
